src/farcasterd/trade_state_machine.rs

```diff
@@ -543,29 +543,20 @@ fn attempt_transition_to_taker_committed(
         acc_addr,
     } = make_offer;
     match (event.request.clone(), event.source.clone()) {
-        (
-            BusMsg::Msg(Msg::TakerCommit(TakeCommit {
-                public_offer: committed_public_offer,
-                swap_id,
-                ..
-            })),
-            ServiceId::Peer(..),
-        ) => {
-            if public_offer == committed_public_offer {
+        (BusMsg::P2p(PeerMsg::TakerCommit(taker_commit)), ServiceId::Peer(..)) => {
+            if public_offer == taker_commit.public_offer {
                 let source = event.source.clone();
-                // FIXME this should go into 1 msg over ctl to avoid race condition and 2 buses
-                // usage
-                let btc_addr_req =
-                    BusMsg::Ctl(Ctl::BitcoinAddress(BitcoinAddress(swap_id, arb_addr)));
-                event.send_msg_service(ServiceId::Wallet, btc_addr_req)?;
-                let xmr_addr_req =
-                    BusMsg::Ctl(Ctl::MoneroAddress(MoneroAddress(swap_id, acc_addr)));
-                event.send_msg_service(ServiceId::Wallet, xmr_addr_req)?;
-                debug!("passing request to walletd from {}", event.source);
-                event.forward_msg(ServiceId::Wallet)?;
+                let TakeCommit { swap_id, .. } = taker_commit;
```

Comment:
Can you reflect this change in the state machine docs for the MakeOffer transition above?

Reply:
Good catch, done in [0cda427](https://github.com/farcaster-project/farcaster-node/pull/742/commits/0cda427679195d763ffe16e27e68cbb48a04be5f)